rust/lance-core/src/io/writer/statistics.rs

```diff
@@ -49,75 +55,385 @@ pub struct StatisticsRow {
+    let mut max_value: Option<&[u8]> = None;
+    let mut null_count: i64 = 0;
+
+    let array_iterator = arrays.iter().map(|x| as_generic_binary_array::<T>(x));
+
+    for array in array_iterator {
+        null_count += array.null_count() as i64;
+        if array.null_count() == array.len() {
+            continue;
+        }
+
+        array.iter().for_each(|value| {
+            if let Some(val) = value {
+                // don't compare full buffers if possible
+                let val = &val[..std::cmp::min(BINARY_PREFIX_LENGTH + 4, val.len())];
+
+                if let Some(v) = min_value {
+                    if let Some(Ordering::Less) = val.partial_cmp(v) {
+                        min_value = Some(val);
+                    }
+                } else {
+                    min_value = Some(val);
+                }
+
+                if let Some(v) = max_value {
+                    if let Some(Ordering::Greater) = val.partial_cmp(v) {
+                        max_value = Some(val);
+                    }
+                } else {
+                    max_value = Some(val);
+                }
+            }
+        });
+    }
+
+    if let Some(v) = min_value {
+        if v.len() > BINARY_PREFIX_LENGTH {
+            min_value = truncate_binary(v, BINARY_PREFIX_LENGTH);
+        }
+    }
+
+    let min_value = min_value.map(|x| x.to_vec());
+    let max_value = if let Some(v) = max_value {
+        if v.len() > BINARY_PREFIX_LENGTH {
+            max_value = truncate_binary(v, BINARY_PREFIX_LENGTH);
+            increment(max_value.unwrap().to_vec())
+        } else {
+            Some(v.to_vec())
+        }
+    } else {
+        None
+    };
+
+    match arrays[0].data_type() {
+        DataType::Binary => StatisticsRow {
+            null_count,
+            min_value: ScalarValue::Binary(min_value),
+            max_value: ScalarValue::Binary(max_value),
+        },
+        DataType::LargeBinary => StatisticsRow {
+            null_count,
+            min_value: ScalarValue::LargeBinary(min_value),
+            max_value: ScalarValue::LargeBinary(max_value),
+        },
+        _ => {
+            todo!()
+        }
+    }
+}
+
+fn get_fixed_size_binary_statistics(arrays: &[&ArrayRef]) -> StatisticsRow {
+    let mut min_value: Option<&[u8]> = None;
+    let mut max_value: Option<&[u8]> = None;
+    let mut null_count: i64 = 0;
+
+    let array_iterator = arrays.iter().map(|x| as_fixed_size_binary_array(x));
+
+    let length = as_fixed_size_binary_array(arrays[0]).value_length() as usize;
+    let length = std::cmp::min(BINARY_PREFIX_LENGTH, length);
+
+    for array in array_iterator {
+        null_count += array.null_count() as i64;
+        if array.null_count() == array.len() {
+            continue;
+        }
+
+        array.iter().for_each(|value| {
+            if let Some(val) = value {
+                // don't compare full buffers if possible
+                let val = &val[..length];
```

Comment:
We don't add 4 here, so either it's okay in the other instances to not, or we should add a comment here why it isn't necessary for this type.

Reply:
We probably should and this is actually not tested at the moment.
Changed the code to add the 4 and will add a test.

Reply:
Added a couple of tests.